lib/awful/screen.lua

```diff
@@ -505,6 +564,100 @@ function screen.object.get_selected_tag(s)
     return screen.object.get_selected_tags(s)[1]
 end
 
+local function normalize(ratios, size)
+    local sum = 0
+
+    for _, r in ipairs(ratios) do
+        sum = sum + r
+    end
+
+    -- Avoid to mutate the input.
+    local ret = {}
+    local sum2 = 0
+
+    for k, r in ipairs(ratios) do
+        ret[k] = (r*100)/sum
+        ret[k] = math.floor(size*ret[k]*0.01)
```

Comment:
Why `*100/0.01`? That can be left out:
```
ret[k] = math.floor(size*r/sum)
```
Also, why does this round down? Doesn't that mean that everything is basically rounded to zero, because this only gets numbers between 0 and 1? Oh, wait... all of this is multiplied with `size`.